sympy/functions/elementary/piecewise.py

```diff
@@ -1019,6 +1019,58 @@ def _eval_rewrite_as_ITE(self, *args, **kwargs):
             last = ITE(c, a, last)
         return _canonical(last)
 
+    def as_non_overlapping(self, var):
+        # Returns Piecewise function with arguments which are non overlapping
+        # Can be used for multivariable as well (as long it can be simplified)
+        # Examples:
+        # >> a = Piecewise((2, x<1-y), (4, x<-1-y),(3, True))
+        # >> a.as_non_overlapping(x+y)
+        #  Piecewise((2, x + y < 1), (3, True))
+        # >> a = Piecewise((2, x>1), (4, x>3),(3, True))
+        # >> a.as_non_overlapping(x)
+        #  Piecewise((2, x > 1), (3, True))
+        from sympy import Intersection, UniversalSet, oo
+
+        newargs = []
+        existing_interval = S.EmptySet
+        _d = Dummy('d')
+        for expr,cond in self.args:
+            # Bring the expression to a single side
+            # x>1-y+z to x-1+y-z>0
```

Comment:
-> `cond.rewrite(Add)`

Reply:
@smichr What will this do?

Reply:
You can try it at an interactive prompt to see that it will do what you are doing: change `x < y` -> `x - y`.

Reply:
@smichr I tried for x<1-y, did not work!